src/printer.js

```diff
@@ -4798,12 +4799,18 @@ function printAstToDoc(ast, options, addAlignmentSize) {
       return genericPrint(path, options, printGenerically, args);
     }
 
-    return comments.printComments(
+    const withComments = comments.printComments(
       path,
       p => genericPrint(p, options, printGenerically, args),
       options,
       args && args.needsSemi
     );
+
+    if (path.getNode() === options.cursorNode) {
```

Comment:
You probably want to put this inside of printComments, there are a few places where we don't print the node but do print the comments.